actors/miner/src/deadlines.rs

```diff
@@ -128,6 +128,64 @@ pub fn deadline_available_for_compaction(
         )
 }
 
+// the distance between from_deadline and to_deadline clockwise in deadline unit.
+fn deadline_distance(policy: &Policy, from_deadline: u64, to_deadline: u64) -> u64 {
+    if to_deadline >= from_deadline {
+        to_deadline - from_deadline
+    } else {
+        policy.wpost_period_deadlines - from_deadline + to_deadline
+    }
+}
+
+// only allow moving to a nearer deadline from current one
+pub fn deadline_available_for_move(
+    policy: &Policy,
+    from_deadline: u64,
+    to_deadline: u64,
+    current_deadline: &DeadlineInfo,
+) -> Result<(), String> {
+    if !deadline_is_mutable(
+        policy,
+        current_deadline.period_start,
+        from_deadline,
+        current_deadline.current_epoch,
+    ) {
+        return Err("cannot move from a deadline when it's not mutable".to_string());
```

Comment:
Please expand the error message to include the relevant details, e.g. "cannot move partition from deadline {}, immutable at epoch {}". Same below.

Reply:
Done.